src/Http/Routing/src/Matching/HttpMethodMatcherPolicy.cs

```diff
@@ -159,7 +159,7 @@ public Task ApplyAsync(HttpContext httpContext, CandidateSet candidates)
             }
         }
 
-        if (needs405Endpoint == true)
+        if (needs405Endpoint == true && methods!=null)
```

Comment:
This doesn't look like the right fix. `CreateRejectionEndpoint` should be updated to allow null-HTTP methods.